lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +44,34 @@ def call
 
         private
 
+        def extract_credentials
+          if @sshkey_path && @token_path
```

Comment:
It's _soooo_ cool that these instance variables are defined and managed in a different file....

Reply:
In a super obtuse way too. Took me a while to figure out how I thought it worked, and even then I wasn't confident till I wrote the tests >.>